Neos.Media/Classes/Eel/AssetsHelper.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Media\Eel;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Eel\ProtectedContextAwareInterface;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Persistence\Exception\InvalidQueryException;
+use Neos\Flow\Persistence\QueryResultInterface;
+use Neos\Media\Domain\Model\AssetCollection;
+use Neos\Media\Domain\Model\AssetInterface;
+use Neos\Media\Domain\Model\Tag;
+use Neos\Media\Domain\Repository\AssetCollectionRepository;
+use Neos\Media\Domain\Repository\AssetRepository;
+use Neos\Media\Domain\Repository\TagRepository;
+
+/**
+ * This is a helper for accessing assets from the media library
+ *
+ * @api
+ */
+class AssetsHelper implements ProtectedContextAwareInterface
+{
+    /**
+     * @Flow\Inject
+     * @var AssetRepository
+     */
+    protected $assetRepository;
+
+    /**
+     * @Flow\Inject
+     * @var TagRepository
+     */
+    protected $tagRepository;
+
+    /**
+     * @Flow\Inject
+     * @var AssetCollectionRepository
+     */
+    protected $assetCollectionRepository;
+
+    /**
+     * @return QueryResultInterface<AssetInterface> | null
+     */
+    public function findByTag(?Tag $tag): ?QueryResultInterface
```

Comment:
Maybe we _should_ allow polymorphism like Christian suggested:

```suggestion
    public function findByTag(Tag|string|null $tag): ?QueryResultInterface
```
?